cluster-autoscaler/simulator/nodes.go

```diff
@@ -44,7 +43,7 @@ func addExpectedPods(nodeInfo *schedulerframework.NodeInfo, scheduledPods []*api
 		}
 		// Add scheduled mirror and DS pods
 		if pod_util.IsMirrorPod(pod) || pod_util.IsDaemonSetPod(pod) {
-			nodeInfo.AddPod(pod)
+			nodeInfo.AddPod(&framework.PodInfo{Pod: pod})
```

Comment:
This type of code makes me nervous - is every place where we pass pod list going to have to be updated to work with DRA? Do we need to never again use pod and just use PodInfo everywhere?

Reply:
Why nervous? It's a fair bit of toil when refactoring now, but doesn't seem risky going forward.

I don't think using PodInfo instead of Pod is strictly needed everywhere in the codebase, but certainly everywhere where the pods are expected to interact with the snapshot. But there's a lot of code that just grabs something from the snapshot, processes the pods and doesn't modify the snapshot. In this case just working on pods is fine IMO.

Reply:
Right, but we need to refactor things like PodListProcessor to take and return list of PodInfos (we definitely interact with snapshot/scheduling in filter out schedulable). At which point we probably want to change it in all processors for consistency. At which point we probably just want to use PodInfo everywhere for consistency.